### PR TITLE
List Endpoint - Search Parameters and Pagination

### PR DESCRIPTION
**Pull request by LBHBReynolds-Carr**
- Created ResidentQueryParam Using existing file from Mosaic API as template
- Implemented Search Parameters and Query
- Implemented pagination
- All tests passing

### CODE REVIEW

File: AcademyResidentInformationApi.Tests/V1/Gateways/AcademyGatewayTests.cs [hunk 1]
```diff
@@ -33,21 +29,17 @@ public void GatewayImplementsBoundaryInterface()
         [Test]
         public void GetAllResidentsIfThereAreNoResidentsReturnsAnEmptyList()
         {
-            _classUnderTest.GetAllResidents().Should().BeEmpty();
+            _classUnderTest.GetAllResidents(0, 20).Should().BeEmpty();
         }
 
         [Test]
         public void GetAllResidentIfThereAreResidentsWillReturnThem()
```
Comment (elcorbs) [inline AcademyResidentInformationApi.Tests/V1/Gateways/AcademyGatewayTests.cs:36]: Seems like this isn't what is being tested here anymore, maybe rename the test?

File: AcademyResidentInformationApi/V1/Boundary/Responses/ResidentInformation.cs [hunk 2]
```diff
@@ -7,6 +7,6 @@ public class ResidentInformation
         public string FirstName { get; set; }
         public string LastName { get; set; }
         public string DateOfBirth { get; set; }
-        public List<Address> AddressList { get; set; }
+        public Address ResidentAddress { get; set; }
```
Comment (elcorbs) [inline AcademyResidentInformationApi/V1/Boundary/Responses/ResidentInformation.cs:10]: Has this change in response been changed in the swagger doc as well?

File: AcademyResidentInformationApi/V1/Gateways/AcademyGateway.cs [hunk 3]
```diff
@@ -17,46 +17,32 @@ public AcademyGateway(AcademyContext academyContext)
...
+            string lastname = null, string postcode = null, string address = null)
         {
-            var addressesWithNoFilters = _academyContext.Addresses
+            var addressesFilteredByPostcode = _academyContext.Addresses
                 .Include(p => p.Person)
+                .Where(a => string.IsNullOrEmpty(address) || a.AddressLine1.ToLower().Replace(" ", "").Contains(StripString(address)))
+                .Where(a => string.IsNullOrEmpty(postcode) || a.PostCode.ToLower().Replace(" ", "").Equals(StripString(postcode)))
+                .Where(a => string.IsNullOrEmpty(firstname) || a.Person.FirstName.ToLower().Replace(" ", "").Equals(StripString(firstname)))
+                .Where(a => string.IsNullOrEmpty(lastname) || a.Person.LastName.ToLower().Replace(" ", "").Equals(StripString(lastname)))
+                .Where(a => a.Person.Id > cursor)
```
Comment (elcorbs) [inline AcademyResidentInformationApi/V1/Gateways/AcademyGateway.cs:29]: I couldn't see any tests around cursor & limit, unless I missed them. Also is the limit being used at all?

**Review by elcorbs — COMMENTED** [review]
Left a few comments but it's looking good. 

There are a few files at the add which look like they are to do with a local session-manger plugin install. Did they get added by accident? Probably best to remove them if so

**Comment by LBHBReynolds-Carr** [comment]
Newer branch accounts for changes from master, pull request closed.

**Pull request closed without merging**